src/components/Search/Search.tsx

```diff
@@ -0,0 +1,62 @@
+import {useRef, useEffect, useState} from 'react';
+
+import {TextInput} from '@gravity-ui/uikit';
+
+interface SearchProps {
+    onChange: (value: string) => void;
+    value?: string;
+    className?: string;
+    debounce?: number;
+    inputRef?: React.Ref<HTMLInputElement>;
+    placeholder?: string;
+}
+
+export const Search = ({
+    onChange,
+    value = '',
+    className,
+    debounce = 200,
+    inputRef,
+    placeholder,
+}: SearchProps) => {
+    const [searchValue, setSearchValue] = useState<string>(value);
+
+    const timer = useRef<number>();
+
+    useEffect(() => {
+        setSearchValue((prevValue) => {
+            if (prevValue !== value) {
+                return value;
+            }
+
+            return prevValue;
+        });
+    }, [value]);
+
+    const onSearchValueChange = (newValue: string) => {
+        setSearchValue(newValue);
+
+        window.clearTimeout(timer.current);
+        timer.current = window.setTimeout(() => {
+            onChange?.(newValue);
+        }, debounce);
+    };
+
+    return (
+        <TextInput
+            controlRef={(node) => {
+                if (typeof inputRef === 'function') {
+                    inputRef(node as HTMLInputElement);
+                } else if (inputRef) {
+                    (inputRef.current as HTMLInputElement) = node as HTMLInputElement;
+                }
+            }}
```

Comment:
i'm not sure if this is useful for this component

Reply:
It was taken from SettingSearch, since it remains as it is, this functionality here is no longer needed

Reply:
Deleted